mobify/platform-scaffold
# Fix Retry button on Offline page

The retry logic for offline functionality was originally built in the App container but the original fetch logic lived in `template.jsx`. This meant that the retry logic wasn't exactly the same as the initial fetch logic, even though they ended up calling the same `fetchAction` that was attached to the current route.

This PR moves the logic that should be shared into a new function in `router.jsx` so that no matter how the `fetchAction` on the route is called, it will handle notifications and offline checks.

 **JIRA**: https://mobify.atlassian.net/browse/WEBDATA-134
 **Linked PRs**: (links to corresponding PRs, optional)

## Changes
- Moved notification cleanup and offline check to `router.jsx`
- Added new `initPage` wrapper for `fetchAction` so that all the shared logic ( ☝️ ) is called no matter ow `fetchAction` is called

## Todos
~~- [ ] (if applicable) analytics events instrumented to measure impact of change~~

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Navigate to a few pages
- Check the "Offline" checkbox on the "Network" tab of dev tools
- Navigate to a page that hasn't been loaded yet. You should see the offline page with a *Retry* button. 
- Tap *Retry* and note that the page is fetched again in the Network tab
- Uncheck the "Offline" checkbox
- Tap *Retry* and note that the page loads and the Offline page is dismissed
